src/main/java/com/code_intelligence/jazzer/agent/RuntimeInstrumentor.kt

```diff
@@ -57,6 +57,17 @@ class RuntimeInstrumentor(
         protectionDomain: ProtectionDomain?,
         classfileBuffer: ByteArray,
     ): ByteArray? {
+        var pathPrefix = ""
+        if (Opt.instrumentOnly && protectionDomain != null) {
+            var jarUrl = protectionDomain?.getCodeSource().getLocation().getFile().toString()
+            if (!jarUrl.isNullOrEmpty()) {
+                var fileName = jarUrl.substring(jarUrl.lastIndexOf('/') + 1, jarUrl.lastIndexOf('.'))
```

Comment:
I'm not sure if this will work in all situations, e.g. if classes are located directly in a directory and not a jar file. Could you please add checks if the returned indices are valid?

Reply:
The code will only work on instrumenting a list of jars that are passed in, directories aren't supported in this check-in. The jar doesn't need to end with .jar though so I did update this to account for that.

Since we switched to using --instrument_only=<list_of_jars> instead of --cp which can contain directories I think this makes sense